domains/index.ts

```diff
@@ -1,25 +1,11 @@
-import { AzureFunction, Context, HttpRequest } from "@azure/functions"
-import { MongoClient } from "mongodb";
-import { MongoDbService } from  "@zero-tech/data-store/src/database/mongo/mongoDbService"
-import { getDbConnectionInfo } from "@zero-tech/data-store/src/database/helpers";
-import * as env from "env-var"
-
-const httpTrigger: AzureFunction = async function (context: Context, req: HttpRequest): Promise<void> {
-
-    const dbUri = env.get("DATABASE_URI").required().asString();
-
-    context.log('HTTP trigger function processed a request.');
-    let dbClient = new MongoClient(dbUri);
-
-    let dbService = new MongoDbService(dbClient, getDbConnectionInfo());
-
-    let results = await dbService.getAllDomains();
-
-    context.res = {
-        // status: 200, /* Defaults to 200 */
-        body: results
-    };
-
+import { AzureFunction, Context, HttpRequest } from "@azure/functions";
+import { doOp, listDomains } from "../services/domainService";
+
+const httpTrigger: AzureFunction = async function (
+  context: Context,
+  req: HttpRequest
+): Promise<void> {
+  await doOp(listDomains, context, req);
```

Comment:
Why pass through this funky doOp function instead of just calling listDomains directly?

You could put a helper function in the domainService to spin up a connection, and call it from the individual functions there.

Reply:
I could see how this looks a little strange coming from C#. But the pattern of passing around functions as arguments is a common and useful pattern in TS/JS. It's very handy for keeping things DRY when you have several small, repeatable pieces of logic that don't happen sequentially. 

This method doesn't only spin up a database connection, but catches all unexpected server errors and handles resource closure of the database.

Reply:
I could see how this looks a little strange coming from C#. But the pattern of passing around functions as arguments is a common and useful pattern in TS/JS. It's very handy for keeping things DRY when you have several small, repeatable pieces of logic that don't happen sequentially. 

This method doesn't only spin up a database connection, but catches all unexpected server errors and handles resource closure of the database.